### PR TITLE
feat: add admission webhook hanlder

### PR DESCRIPTION
**Pull request by ComradeProgrammer**
feat: add admission webhook hanlder

### CODE REVIEW

**Comment by hsluoyz** [comment]
@sagilio

File: internal/model/model_adaptor_loader.go [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+	"github.com/casbin/casbin/v2/persist"
+	"github.com/casbin/k8s-gatekeeper/pkg/crdadaptor"
+	"github.com/casbin/k8s-gatekeeper/pkg/generated/clientset/versioned"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/homedir"
+)
+
+type ModelAdaptorLoader struct {
```
Comment (sagilio) [inline internal/model/model_adaptor_loader.go]: Why not named `ModelLoader` directly

File: internal/model/synchronized_enforcers.go [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+	casbin "github.com/casbin/casbin/v2"
+	admission "k8s.io/api/admission/v1"
+)
+
+type EnforcerWrapper struct {
+	Enforcer  *casbin.Enforcer
+	ModelName string
+}
+
+type SynchronizedEnforcerList struct {
```
Comment (sagilio) [inline internal/model/synchronized_enforcers.go]: This name is too long for the golang code style

File: internal/model/model_adaptor_loader.go [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+	var err error
+	if isExternalClient {
+		err = res.establishExternalClient()
+	} else {
+		err = res.establishInternalClient()
+	}
+	return res, err
+}
+
+func (m *ModelAdaptorLoader) GetModelAndAdaptors() ([]ModelAdaptorPair, error) {
```
Comment (sagilio) [inline internal/model/model_adaptor_loader.go]: I think this method or ModelLoader maybe has a better way to implement it, the model and adapter should not be closed like this.

File: internal/model/synchronized_enforcers.go [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+	}
+	res := &SynchronizedEnforcerList{
+		Enforcers: make([]*EnforcerWrapper, 0),
+		loader:    loader,
+	}
+	//load all enabled models and rules
+	res.loadEnforcer()
+	//start auto sync for loaders
+	go func() {
+		for {
```
Comment (sagilio) [inline internal/model/synchronized_enforcers.go:44]: select and ticker may be better than this.

File: internal/model/synchronized_enforcers.go [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+func (s *SynchronizedEnforcerList) loadEnforcer() {
+	s.Lock()
+	defer s.Unlock()
+
+	modelAndAdptors, err := s.loader.GetModelAndAdaptors()
+	if err != nil {
+		log.Printf("error: %s", err.Error())
+		return
+	}
+	s.Enforcers = s.Enforcers[:0]
```
Comment (sagilio) [inline internal/model/synchronized_enforcers.go:77]: What is the purpose of this line?

**Review by sagilio — COMMENTED** [review]
A lot of places need to be carefully polished

File: internal/model/synchronized_enforcers.go (hunk 5 above)
Comment (ComradeProgrammer) [reply internal/model/synchronized_enforcers.go:77]: resize this slice to empty, clear all existing enforcers

Comment (sagilio) [reply internal/model/synchronized_enforcers.go:77]: why not make a new slice?

**Review by sagilio — APPROVED** [review]
LGTM

**Pull request merged**